### PR TITLE
Convert from create-react-class to es6 class

### PR DESCRIPTION
**Pull request by ChrisNLott**
Remove the `create-react-class` dep

### CODE REVIEW

File: src/AutosizeInput.js [hunk 1]
```diff
@@ -152,7 +158,28 @@ const AutosizeInput = createClass({
 				}
 			</div>
 		);
-	},
-});
+	};
+}
+
+AutosizeInput.propTypes = {
```
Comment (RyanZim) [inline src/AutosizeInput.js]: Can't you do `static propTypes =` in the class definition?

Comment (ChrisNLott) [inline src/AutosizeInput.js]: You can, but I like having the propTypes next to the defaultProps. Personal preference though

Comment (ChrisNLott) [inline src/AutosizeInput.js]: Plus eslint was complaining about `static propTypes =`

Comment (RyanZim) [inline src/AutosizeInput.js]: OK, just wondered.

Comment (ChrisNLott) [inline src/AutosizeInput.js]: Yeah i don't imagine there are any performance improvements from using the `static` syntax

**Comment by jochenberger** [comment]
What needs to be done to get this in? Maybe change the PR to not include generated files (`lib/*`, `dist/*`, and `example/dist/*`)?

**Comment by tshelburne** [comment]
@JedWatson Do you have any idea of timing on this one? We are working on removing deprecation warnings while preparing for React v16, and this is the last piece left. Not intending to pressure, just trying to get a schedule put together.

**Comment by jochenberger** [comment]
@ChrisNLott, any chance you can rebase this on master? @JedWatson seems to be having a run. :-)

**Comment by JedWatson** [comment]
Don't worry, I'm on it :)

**Comment by JedWatson** [comment]
Done! #103

**Comment by JedWatson** [comment]
Thanks for the work on this @ChrisNLott, I wanted to upgrade the build as well (based on what we just shipped for react-select) so we can use modern babel plugins so it's a more comprehensive rewrite now.

**Comment by tshelburne** [comment]
Thanks!

**Pull request closed without merging**